src/entities/addLiquidityBoosted/index.ts

```diff
@@ -0,0 +1,187 @@
+// A user can add liquidity to a boosted pool in various forms. The following ways are
+// available:
+// 1. Unbalanced - addLiquidityUnbalancedToERC4626Pool
+// 2. Proportional - addLiquidityProportionalToERC4626Pool
+import { TokenAmount } from '@/entities/tokenAmount';
+
+import { Permit2 } from '@/entities/permit2Helper';
+
+import { getAmountsCall } from '../addLiquidity/helpers';
+
+import { PoolState, PoolStateWithUnderlyings } from '@/entities/types';
+
+import {
+    AddLiquidityBaseBuildCallInput,
+    AddLiquidityBaseQueryOutput,
+    AddLiquidityBuildCallOutput,
+    AddLiquidityInput,
+    AddLiquidityKind,
+} from '../addLiquidity/types';
+
+import { doAddLiquidityUnbalancedQuery } from './doAddLiquidityUnbalancedQuery';
+import { doAddLiquidityProportionalQuery } from './doAddLiquidityPropotionalQuery';
+import { Token } from '../token';
+import { BALANCER_COMPOSITE_LIQUIDITY_ROUTER } from '@/utils';
+
+import { getValue } from '@/entities/utils/getValue';
+import { encodeFunctionData } from 'viem';
+import { balancerCompositeLiquidityRouterAbi, balancerRouterAbi } from '@/abi';
+
+import { Hex } from '@/types';
+
+export class AddLiquidityBoostedV3 {
+    async query(
+        input: AddLiquidityInput,
+        poolState: PoolState | PoolStateWithUnderlyings,
+    ): Promise<AddLiquidityBaseQueryOutput> {
+        // Technically possible to pass singleToken adds here due to type
```

Comment:
Can change type to be 
`AddLiquidityUnbalancedInput
    | AddLiquidityProportionalInput;`